mod-source-record-manager-server/src/main/java/org/folio/dao/JobExecutionDaoImpl.java

```diff
@@ -286,10 +303,16 @@ private Tuple mapToTuple(JobExecution jobExecution) {
 
   private JobExecutionDtoCollection mapToJobExecutionDtoCollection(RowSet<Row> rowSet) {
     JobExecutionDtoCollection jobCollection = new JobExecutionDtoCollection().withTotalRecords(0);
-    for (Row row : rowSet) {
+    rowSet.iterator().forEachRemaining(row -> {
       jobCollection.getJobExecutions().add(mapRowToJobExecutionDto(row));
       jobCollection.setTotalRecords(row.getInteger(TOTAL_COUNT_FIELD));
-    }
+    });
+    return jobCollection;
+  }
+
+  private JobProfileInfoCollection mapRowToJobProfileInfoCollection(RowSet<Row> rowSet) {
+    JobProfileInfoCollection jobCollection = new JobProfileInfoCollection().withTotalRecords(rowSet.rowCount());
```

Comment:
rowSet.rowCount return total number of unique profiles, not only returned in response for current limit, offset, is it correct?

Reply:
Hi @SerhiiNosko. unfortunately, only the number including offset and limit.

Reply:
Basically its not correct contract for pagination, because consumers of the paginated endpoint need to know how many requests need to done to fetch all records.
Can you please check how pagination implemented in other endpoints?

Reply:
I hadn't considered that point. It will be fixed. Thank you)